avm/src/vm/memory.rs

```diff
@@ -154,11 +153,16 @@ impl HandlerMemory {
   /// that cannot be performed on an object within an Arc. As a result,
   /// all the forked children drop their reference to the parent, we take
   /// the parent out of the Arc after parallel work is done and then join on its children.
-  pub fn drop_parent(self: &mut HandlerMemory) {
-    if self.parent.is_some() {
-      let arc = self.parent.take();
-      drop(arc);
+  pub fn drop_parent(mut self: Arc<HandlerMemory>) -> Arc<HandlerMemory> {
+    let hm = Arc::get_mut(&mut self).expect("unable to drop parent HM: dangling pointer");
+    if let Some(parent) = hm.parent.take() {
+      drop(parent);
     }
+    self
+  }
+
+  pub(crate) async fn _drop_parent(self: Arc<HandlerMemory>) -> Arc<HandlerMemory> {
```

Comment:
can we call this `drop_parent_async` like I had mentioned during the pair program?